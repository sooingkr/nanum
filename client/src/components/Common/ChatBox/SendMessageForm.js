import React, { Component } from "react";
import { reduxForm, Field } from 'redux-form';
import { Button, Form, FormGroup, FormControl} from "react-bootstrap";

const InputMessageBoxView = ({ input, pristine, submitting }) => (
  <FormGroup>
    <FormControl type="text" placeholder="네, 작업장입니다|" {...input}/>
    <Button className="btn-send" type="submit" disabled={pristine || submitting}>
      <i className="fa fa-paper-plane" aria-hidden="true"/>
    </Button>
  </FormGroup>
);

class SendMessageFormView extends Component {

  componentWillMount() {
    const { typingMessage, pristine } = this.props;
    const isTyping = typingMessage(!pristine);

    return isTyping;
  }

  render() {
    const { handleSubmit, submitting, className, isTyping, pristine } = this.props;

    // notify that user is typing
    // typingMessage(!pristine);

    return (
      <Form onSubmit={handleSubmit} className={className}>
        <Field name="message" component={ InputMessageBoxView } {...{ pristine, submitting, isTyping }}/>
      </Form>
    );
  }
}

const SendMessageForm = reduxForm({
  form: 'SendMessageForm',
})(SendMessageFormView);

export default SendMessageForm;