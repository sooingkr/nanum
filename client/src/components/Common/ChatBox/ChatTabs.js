import React, { Component } from "react";
import { isEmpty } from "lodash";

import { Tabs, Tab } from "react-bootstrap";

import ChatBox from "../../Common/ChatBox/ChatBox";
import SendMessageForm from "../../Common/ChatBox/SendMessageForm";

class TabComponent extends Component {

  render() {

    const { tabData: { messages, isTyping, typingMessage, formData } } = this.props;
    // const { messages, toggleChatBox, typingMessage, formData, isTyping } = this.props;

    return (
      <Tabs defaultActiveKey={1} id="uncontrolled-tab-example">
        <Tab eventKey={1} title="안전먹거리">
          <div className="chat-messages">
            <div className="block__chat-messages">
              {
                !(isEmpty(messages)) ?
                  messages.map((message, index) =>
                    <ChatBox key={index} index={index} message={message} isTyping={isTyping}></ChatBox>
                  ) : null
              }
            </div>
            <div className="chat-messages__form">
              <SendMessageForm className="chat-form__safefood" onSubmit={formData} typingMessage={typingMessage}/>
            </div>
          </div>
        </Tab>
        <Tab eventKey={2} title="HACCP기술상담">
          <div className="chat-messages">
            <div className="block__chat-messages">
              {
                !(isEmpty(messages)) ?
                  messages.map((message, index) =>
                    <ChatBox key={index} index={index} message={message} isTyping={isTyping}></ChatBox>
                  ) : null
              }
            </div>
            <div className="chat-messages__form">
              <SendMessageForm className="chat-form__haccp" onSubmit={formData} typingMessage={typingMessage}/>
            </div>
          </div>
        </Tab>
      </Tabs>
    );
  }
}

export default TabComponent;