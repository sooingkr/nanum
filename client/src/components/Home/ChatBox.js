import React, { Component } from "react";
import { Image } from "react-bootstrap";

import { isEmpty } from "lodash";
import { PropTypes } from 'prop-types';

import "./ChatBox.scss";

class ChatBox extends Component {
  render() {
    const { message: { index, content, admes } } = this.props;

    return (
      <div className="chat-messages__text" key={ index }>
        { isEmpty( content ) ? null
          :
          <div className="chat-messages__user">
            <div className="image-block pull-left">
              <Image src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/User_icon_2.svg/220px-User_icon_2.svg.png"
                circle/></div>
            <div className="chat-messages__box user">{ content }</div>
          </div>
        }

        { isEmpty( admes ) ? null
          :
          <div className="chat-messages__box supporter">
            { admes }
          </div>
        }
      </div>
    );
  }
}

ChatBox.propTypes = {
  message: PropTypes.shape({
    index: PropTypes.number,
    content: PropTypes.string.isRequired,
    admes: PropTypes.string
  }).isRequired
}

export default ChatBox;