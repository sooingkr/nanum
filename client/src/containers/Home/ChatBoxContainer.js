import React, { Component } from "react";
import { connect } from 'react-redux';

import { Row, Col, Button, Collapse, Well, Form, FormGroup, FormControl, Image } from "react-bootstrap";
import ChatBox from "../../components/Home/ChatBox.js";

import { PropTypes } from "prop-types";
import { homeDuck } from "./HomeDuck";

class ChatBoxContainer extends Component {

  componentWillMount() {
  }

  render() {
    const { userId, messages, openChatBox, toggleChatBox } = this.props;

    return (
      <div className="chat-box">
        <div className="block__chat-box">
          <Button className="btn-open" title="무엇을 도와드릴까요?" onClick={() => { toggleChatBox(); }}>
            무엇을 도와드릴까요? <i className="fa fa-arrow-circle-right" aria-hidden="true"></i>
          </Button>

          <Collapse in={openChatBox}>
            <div>
              <Well>
                <div className="chat-box__title">
                  <Row>
                    <Col xs={8} xsPush={2}><h5 className="chat-box__title-text text-center"><i
                      className="fa fa-comment-o"></i><span>기술상담</span></h5></Col>
                    <Col xs={1} xsPush={1}><Button className="btn-close" onClick={() => { toggleChatBox(); }}><span className="line"></span></Button></Col>
                  </Row>
                </div>
                <div className="chat-messages">
                  <div className="block__chat-messages">
                    {
                      messages.map((message, index) =>
                        <ChatBox key={index} message={message}></ChatBox>
                      )
                    }
                  </div>
                  <div className="chat-messages__form">
                    <Form>
                      <FormGroup>
                        <FormControl type="text" placeholder="네, 작업장입니다|"/>
                        <Button className="btn-send"><i className="fa fa-paper-plane" aria-hidden="true"></i></Button>
                      </FormGroup>
                    </Form>
                  </div>
                </div>
              </Well>
            </div>
          </Collapse>
        </div>
        <div className="image-block user-icon">
          <Image
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/User_icon_2.svg/220px-User_icon_2.svg.png"
            circle/>
        </div>
      </div>
    );
  }
}

ChatBoxContainer.propTypes = {
  userId: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  messages: PropTypes.array.isRequired
};

const mapStateToProps = state => {
  const homeState = state[homeDuck.storeName];
  return {
    openChatBox: homeState.openChatBox
  };
};

const mapDispatchToProps = {
  toggleChatBox: homeDuck.actions.toggleChatBox
};

export default connect(mapStateToProps, mapDispatchToProps)(ChatBoxContainer);