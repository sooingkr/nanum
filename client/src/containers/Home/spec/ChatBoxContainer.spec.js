import React from 'react';
import { shallow, mount } from 'enzyme';

import { ChatBoxContainer } from '../ChatBoxContainer';
import ChatBox from "../../../components/Home/ChatBox";

describe('ChatBoxContainer component', () => {
  let props;
  let mountedChatBox;

  const chatBoxContainer = () => {
    if (!mountedChatBox) {
      mountedChatBox = mount(
        <ChatBoxContainer {...props}/>
      )
    }
    return mountedChatBox;
  }

  beforeEach(() => {
    props = {
      userId: '1',
      messages: [
        {
          id: '1',
          content: '',
          admes: 'HACCP 교육 일정을 알려주세요.'
        },
        {
          id: '2',
          content: '무엇을 도와드릴까요?',
          admes: 'HACCP 교육 일정을 알려주세요.'
        },
        {
          id: '3',
          content: '무엇을 도와드릴까요?',
          admes: ''
        },
        {
          id: '1',
          content: '무엇을 도와드릴까요?',
          admes: 'HACCP 교육 일정을 알려주세요.'
        },
        {
          id: '1',
          content: '무엇을 도와드릴까요?',
          admes: 'ad message 2'
        },
        {
          id: '1',
          content: '무엇을 도와드릴까요?',
          admes: ''
        }
      ],
    };

    mountedChatBox = undefined;
  });

  it('renders without crashing', () => {
    shallow(<ChatBoxContainer {...props} />);
  });

  describe('render ChatBox component', () => {
    it ("always render ChatBox component", () => {
      expect(chatBoxContainer().find(ChatBox).length).toBeGreaterThanOrEqual(1);
    });
  });

  describe('<form/> element', () => {
    it('should have a `<form>` element in ChatBoxContainer', () => {
      expect(chatBoxContainer().find('form').length).toBe(1);
    });

    describe('<input/> element', () => {
      it('`<input>` element should be of type `text`', () => {
        expect(chatBoxContainer().find('form').childAt(0).props().type).toBe('text');
      });

      it('`<input>` element value should be empty', () => {
        expect(
          chatBoxContainer().find('form').childAt(0).props().value).toBe('');
      });

    });

    describe('<button/> element', () => {
      it('should have a `<button>` element', () => {
        const form = chatBoxContainer().find('form');
        expect(form.find("button").length).toBe(1);
      });
    });
  });

});




