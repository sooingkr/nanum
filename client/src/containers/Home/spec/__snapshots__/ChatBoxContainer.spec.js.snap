// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBoxContainer component should render <ChatBoxContainer /> components 1`] = `
<div
  className="chat-box"
>
  <div
    className="block__chat-box"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-open"
      disabled={false}
      title="무엇을 도와드릴까요?"
    >
      무엇을 도와드릴까요? 
      <i
        aria-hidden="true"
        className="fa fa-arrow-circle-right"
      />
    </Button>
  </div>
  <div
    className="image-block user-icon"
  >
    <Image
      alt="user-image"
      bsClass="img"
      circle={true}
      responsive={false}
      rounded={false}
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/User_icon_2.svg/220px-User_icon_2.svg.png"
      thumbnail={false}
    />
  </div>
</div>
`;
